dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/comparator/AscendingPeriodComparator.java

```diff
@@ -46,13 +47,12 @@
     @Override
     public int compare( Period period1, Period period2 )
     {
-        int startDateCompared = period1.getStartDate().compareTo( period2.getStartDate() );
-        
-        if ( startDateCompared != 0 )
-        {
-            return startDateCompared;
-        }
-        
-        return period1.getEndDate().compareTo( period2.getEndDate() );
+        PeriodType a = period1.getPeriodType();
+        PeriodType b = period2.getPeriodType();
+
+        int freqCompare = Integer.compare( a.getFrequencyOrder(), b.getFrequencyOrder() );
+        int nameCompare = a.getName().compareTo( b.getName() );
+
+        return freqCompare == 0 ? nameCompare == 0 ?  period1.getStartDate().compareTo(period2.getStartDate() ) : nameCompare : freqCompare;

```

Comment:
Could you write this as if .. else or add paranthesis to make it easier to read?